ale_linters/haskell/hie.vim

```diff
@@ -0,0 +1,29 @@
+" Author: Luxed <devildead13@gmail.com>
+" Description: A language server for Haskell
+
+call ale#Set('haskell_hie_executable', 'hie')
+
+function! ale_linters#haskell#hie#GetExecutable(buffer) abort
+    return ale#Var(a:buffer, 'haskell_hie_executable')
+endfunction
+
+function! ale_linters#haskell#hie#GetCommand(buffer) abort
+    let l:executable = ale_linters#haskell#hie#GetExecutable(a:buffer)
+    let l:arguments = '--lsp'
+
+    return ale#Escape(l:executable) . ' ' . l:arguments
+endfunction
```

Comment:
Try writing just `'command': '%e --lsp',` below. I added support for a special `%e` format marker for adding in the escaped executable recently.